scan/lib/scan/test_command_generator.rb

```diff
@@ -49,7 +50,9 @@ def actions
         actions = []
         actions << :clean if config[:clean]
         actions << :build unless config[:skip_build]
-        actions << :test
+        actions << "build-for-testing" if config[:build_for_testing]
+        actions << :test unless config[:test_without_building] || config[:build_for_testing]
```

Comment:
The way these actions are being managed is a little cumbersome. I'd be inclined to design it such that `scan` supports a new `action` option, which defaults to "test". Other possible values would be "build-for-testing" or "test-without-building" to match the actions passed to xcodebuild.

It seems wise to move closer to the 🍎  rather than have scan's UI diverge further from what's happening under the hood.